BlueprintUI/Sources/Internal/CacheTree.swift

```diff
@@ -23,18 +26,29 @@ extension CacheTree {
     /// Convenience method to get a cached size, or compute and store one if it is not in the cache.
     func get(_ constraint: SizeConstraint, orStore calculation: (SizeConstraint) -> CGSize) -> CGSize {
 
-        if let size = self[constraint] {
+        /// Due to various math upstream, we can end up with measurements like 210.0 getting
+        /// turned into numbers like 209.99999997 due to loss of precision.
+        ///
+        /// Because we cache by exact values in this cache,
+        /// let's round the key to the nearest screen-scale pixel to avoid a cache miss.
+        ///
+        /// ⚠️ Important! We **do not** want to round the constraint itself, just the key.
+        /// Blueprint allows (and expects) unrounded values – they're rounded / snapped during the view layout pass.
+
+        let key = constraint.roundToNearestPixel(with: screenScale)
```

Comment:
Since we don't actually update the constraint, how important is the `screenScale` rounding?